### PR TITLE
Git Commits For Non-Git Users

### PR DESCRIPTION
**Pull request by ThoughtbotThom**
Adding a guide and link from contributing for those not familiar with git to contribute to TIL

### CODE REVIEW

File: GitCommitsForNonGitUsers.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
+#Git Commits For Non-Git Users
+
+This is a guide to aid others not familiar with git to contribute to our TIL repo!
```
Comment (mike-burns) [inline GitCommitsForNonGitUsers.md]: No exclamation mark.

File: GitCommitsForNonGitUsers.md [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+####FIRST TIME SETUP
```
Comment (mike-burns) [inline GitCommitsForNonGitUsers.md]: Titlecase. Put a space after the `#`. Only two `#`.

File: GitCommitsForNonGitUsers.md [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+* copy clone to clipboard  
+![](http://images.thoughtbot.com/TIL/copyClone_s.jpg)
+* in Terminal (as are all `commands` that follow) `git clone <PASTE CLIPBOARD>`
```
Comment (mike-burns) [inline GitCommitsForNonGitUsers.md]: I am entertained, but do not disagree, with the assumption that people who don't know git use OS X.

File: GitCommitsForNonGitUsers.md [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+* write title & message  
+* `ESC :wq` to write-quit  
+* `git push origin <BRANCHNAME>`  
+* Compare & Pull Request on website
+
+####TO CHANGE
+* `git status` to see changed files in red
+* `git add .`
+* `git status` to see updated files in green
+* `git commit` ( -m "MESSAGE" to skip Vim)
```
Comment (mike-burns) [inline GitCommitsForNonGitUsers.md]: Do not use `-m`.

File: GitCommitsForNonGitUsers.md [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+* `git push origin <BRANCHNAME>`
+
+####TO SQUISH
```
Comment (mike-burns) [inline GitCommitsForNonGitUsers.md]: Squash.

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -10,6 +10,8 @@ Create a new directory if no existing one fits.
 
 Open a pull request for review.
 
+For help submitting, please review our [Git Commits for Non-Git Users](/GitCommitsForNonGitUsers.md) guide.
```
Comment (mike-burns) [inline CONTRIBUTING.md]: This filename is crazy. How about just inlining the contents of that file into `CONTRIBUTING.md`.

File: GitCommitsForNonGitUsers.md [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+![](http://images.thoughtbot.com/TIL/copyClone_s.jpg)
+* in Terminal (as are all `commands` that follow) `git clone <PASTE CLIPBOARD>`
+
+####TO SUBMIT
+* `git pull origin master`  
+* `git checkout -b <BRANCHNAME>`  
+* add files/folders locally
+* `git add .` to add everything in cur dir  
+* `git commit`  
+* write title & message
```
Comment (weiland) [inline GitCommitsForNonGitUsers.md]: In most cases the user has to press `i` to switch to insert mode or am i wrong?

Comment (mike-burns) [inline GitCommitsForNonGitUsers.md]: Is $VISUAL set to vim by default on OS X? Can we recommend something else?

    env GIT_EDITOR=TextEdit git commit

**Comment by ThoughtbotThom** [comment]
Greg, there's a screenshot attached of where to click on UI

File: GitCommitsForNonGitUsers.md [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+
+## TO SQUASH
+* `git fetch origin`
+* `git rebase -i origin/master`
+* squash to 1 in vim; `:wq`
+* reword commit messages
+
+## TO MERGE
+* `git log`
+* `git push -f origin <BRANCHNAME>`
```
Comment (JoelQ) [inline GitCommitsForNonGitUsers.md]: I think this should go in the "To squash" section. You don't need to force push if you haven't squashed

File: GitCommitsForNonGitUsers.md [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
+* `git push origin <BRANCHNAME>`
+
+## TO SQUASH
+* `git fetch origin`
+* `git rebase -i origin/master`
+* squash to 1 in vim; `:wq`
+* reword commit messages
+
+## TO MERGE
+* `git log`
```
Comment (JoelQ) [inline GitCommitsForNonGitUsers.md]: Maybe add a note saying this is to view the list of commits on the current branch?

**Pull request merged**